processor/src/errors.rs

```diff
@@ -21,10 +26,95 @@ pub enum ExecutionError {
     NttDomainSizeTooSmall(u64),
     NttDomainSizeNotPowerof2(u64),
     InvalidStackDepthOnReturn(usize),
-    InterpolationResultSizeTooBig(u64),
+    InterpolationResultSizeTooBig(usize, usize),
     NotBinaryValue(Felt),
     NotU32Value(Felt),
     ProverError(ProverError),
     SyscallTargetNotInKernel(Digest),
-    UnexecutableCodeBlock(CodeBlock),
+    UnexecutableProxyBlock(CodeBlock),
 }
+
+impl Display for ExecutionError {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), core::fmt::Error> {
+        match self {
+            ExecutionError::AdviceKeyNotFound(key) => {
+                write!(fmt, "Value {key:?} not presented on the advice map, can't write it to tape")
+            }
+            ExecutionError::AdviceSetLookupFailed(reason) => {
+                write!(fmt, "Advice set lookup failed, reason {reason}")
+            }
+            ExecutionError::AdviceSetNotFound(name) => {
+                write!(fmt, "Advice set {name:?} not found")
+            }
+            ExecutionError::AdviceSetUpdateFailed(reason) => {
+                write!(fmt, "Advice set update failed, reason {reason}")
+            }
+            ExecutionError::AdviceTapeReadFailed(step) => {
+                write!(fmt, "Advice tape read failed at step {step}")
+            }
+            ExecutionError::CodeBlockNotFound(digest) => {
+                write!(
+                    fmt,
+                    "Execution of call instruction failed, function body {digest:?} not found"
```

Comment:
nit: I'd change this to: "Failed to execute code block with root {digest}; the block could not be found"

Also, it would be great if `digest` got printed out as a hex string.